projects/ad2s1210_iio/app/app_config_mbed.c

```diff
@@ -0,0 +1,78 @@
+/***************************************************************************//**
+ *   @file    app_config_mbed.c
+ *   @brief   Application configurations module for Mbed platform
+********************************************************************************
+ * Copyright (c) 2023 Analog Devices, Inc.
+ * Copyright (c) 2023 BayLibre, SAS.
+ * All rights reserved.
+ *
+ * This software is proprietary to Analog Devices, Inc. and its licensors.
+ * By using this software you agree to the terms of the associated
+ * Analog Devices Software License Agreement.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include <stdbool.h>
+#include "app_config.h"
+#include "app_config_mbed.h"
+
+/******************************************************************************/
+/************************ Macros/Constants ************************************/
+/******************************************************************************/
+
+/******************************************************************************/
+/******************** Variables and User Defined Data Types *******************/
+/******************************************************************************/
+
+/* UART Mbed platform specific init parameters */
+struct mbed_uart_init_param mbed_uart_extra_init_params = {
+	.uart_tx_pin = UART_TX,
+	.uart_rx_pin = UART_RX,
+#if defined(USE_PHY_COM_PORT)
+	.is_console_stdio_port = false,
+#else
+	.is_console_stdio_port = true,
+#endif
+};
+
+/* VCOM Mbed platform specific init parameters */
+struct mbed_uart_init_param mbed_vcom_extra_init_params = {
+	.vendor_id = VIRTUAL_COM_PORT_VID,
+	.product_id = VIRTUAL_COM_PORT_PID,
+	.serial_number = VIRTUAL_COM_SERIAL_NUM,
+#if defined(USE_VIRTUAL_COM_PORT)
+	.is_console_stdio_port = false,
+#else
+	.is_console_stdio_port = true,
+#endif
+};
+
+/* SPI Mbed platform specific parameters */
+struct mbed_spi_init_param mbed_spi_extra_init_params = {
+	.spi_clk_pin = SPI_SCK,
+	.spi_miso_pin = SPI_HOST_SDI,
+	.spi_mosi_pin = SPI_HOST_SDO,
+	.use_sw_csb = true
```

Comment:
any particular reason of using software CSB? As per our experience with Mbed, software CSB adds-up some extra dead time during data transfer, whereas with hardware CSB this dead time is less compared to software CSB. This is expected as we are explicitly toggling CS pin when it is configured as software CSB, while when it is configured as h/w CSB, toggling of CSB happen within Mbed layer, so this extra overhead is less.

Reply:
i need to try that.